### PR TITLE
feat(components): Allow GridCell' s to only be usable in Grids

### PR DESCRIPTION
**Pull request by MichaelParadis**
### Added

- 

### Changed

- Made it so `Grid.Cell` can only be used inside of a grid component

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

- Install pre-release.
- Add `Grid.Cell` that is not wrapped in a `Grid`. Verify an error occurs
- Add `Grid.Cell` that is wrapped in a `Grid`. Verify an error doesn't occur.

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by darryltec** [comment]
Oooh, that's a cool approach. What happens when you do it like so?

```html
<Grid>
  <Grid.cell>
    <Grid.cell> </Grid.cell>
  </Grid.cell>
<Grid>
```

would it accept it and not throw an error?

**Comment by JeffMarvin** [comment]
@MichaelParadis Is there any functional difference/benefit to using this approach vs. the `Chip.tsx` with `useAssert` hook that we have used in the Chips component? (https://github.com/GetJobber/atlantis/blob/master/packages/components/src/Chips/Chip.tsx)

I find that approach a bit more understandable on the surface, so I'm curious what we gain with this approach. :)

**Comment by MichaelParadis** [comment]
@darryltec @JeffMarvin Updated to use the Chips style of handling it

File: packages/components/src/Grid/GridCell.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Only need the component to pass in the data as a react children
+
+import { useAssert } from "@jobber/hooks/useAssert";
+import React, { PropsWithChildren } from "react";
+import { GridCellProps } from "./InternalGridCell";
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function GridCell(props: PropsWithChildren<GridCellProps>) {
+  // Throw error when <Chip /> gets used outside of the <Chips />
```
Comment (darryltec) [inline packages/components/src/Grid/GridCell.tsx:9]: Oopsies 

```suggestion
  // Throw error when <Grid.Cell /> gets used outside of the <Grid />
```

File: packages/components/src/Grid/Grid.tsx [hunk 2]
```diff
@@ -36,7 +37,9 @@ export function Grid({
 
   return (
     <div data-testid={GRID_TEST_ID} className={classnames}>
-      {children}
+      {React.Children.map(children, gridCell => {
+        return <InternalGridCell {...gridCell.props} />;
```
Comment (darryltec) [inline packages/components/src/Grid/Grid.tsx:41]: nit

```suggestion
      {React.Children.map(children, { props } => {
        return <InternalGridCell {...props} />;
```

**Review by darryltec — CHANGES_REQUESTED** [review]
1 very minor oops

File: packages/components/src/Grid/GridCell.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// Only need the component to pass in the data as a react children
+
+import { useAssert } from "@jobber/hooks/useAssert";
+import React, { PropsWithChildren } from "react";
+import { GridCellProps } from "./InternalGridCell";
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function GridCell(props: PropsWithChildren<GridCellProps>) {
```
Comment (darryltec) [inline packages/components/src/Grid/GridCell.tsx:8]: underscore did not work for this eh?

```suggestion
export function GridCell(_: Props>) {
```

**Review by darryltec — APPROVED** [review]
:shipit:

**Pull request merged**